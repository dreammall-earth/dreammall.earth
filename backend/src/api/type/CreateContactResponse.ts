// eslint-disable-next-line import/no-namespace
import * as http from 'http'

// eslint-disable-next-line import/no-namespace
import * as SibApiV3Sdk from '@getbrevo/brevo'

export type CreateContactResponse = {
  response: http.IncomingMessage
  body: SibApiV3Sdk.CreateUpdateContactModel
}
