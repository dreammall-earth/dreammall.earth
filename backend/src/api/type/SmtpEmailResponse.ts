// eslint-disable-next-line import/no-namespace
import * as http from 'http'

// eslint-disable-next-line import/no-namespace
import * as SibApiV3Sdk from '@getbrevo/brevo'

export type SmtpEmailResponse = {
  response: http.IncomingMessage
  body: SibApiV3Sdk.CreateSmtpEmail
}
