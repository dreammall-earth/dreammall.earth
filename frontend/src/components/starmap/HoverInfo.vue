<template>
  <div v-show="!!props.data" ref="hoverRef" class="hover-info" :style="style">
    <div class="info-box pa-2">
      <v-avatar :size="25" class="avatar mb-2 text-font">
        <span>{{ props.data && getInitials(props.data.name) }}</span>
      </v-avatar>
      <h3>{{ props.data?.name }}</h3>
    </div>
    <div v-show="showMoreButton" class="mt-2 d-flex align-center justify-center">
      <button @click="showMore">
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="28"
          height="28"
          viewBox="0 0 28 28"
          fill="none"
        >
          <g opacity="0.75">
            <circle cx="14" cy="14" r="14" fill="#3D4753" />
            <circle cx="14" cy="14" r="13.5" stroke="#979797" stroke-opacity="0.3" />
          </g>
          <circle cx="7" cy="14" r="2" transform="rotate(-90 7 14)" fill="#CCCCCC" />
          <circle cx="14" cy="14" r="2" transform="rotate(-90 14 14)" fill="#CCCCCC" />
          <circle cx="21" cy="14" r="2" transform="rotate(-90 21 14)" fill="#CCCCCC" />
        </svg>
      </button>
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed } from 'vue'

import { getInitials } from '#src/utils/getInitials'

import type { UserWithProfile } from '#stores/userStore'

const props = defineProps<{
  data: UserWithProfile | null
  x: number
  y: number
  showMoreButton: boolean
}>()

const emit = defineEmits<{
  (e: 'show-more'): void
}>()

const showMore = () => emit('show-more')

const coordinates = computed(() => ({
  x: props.x,
  y: props.y,
}))

const style = computed(() => ({
  left: `${coordinates.value.x - 50}px`,
  top: `${coordinates.value.y - 50}px`,
}))
</script>

<style scoped>
.hover-info {
  position: fixed;
  z-index: 100;
}

.info-box {
  width: 100px;
  height: 100px;
  font-size: 10px;
  color: #f5f5f5;
  text-align: center;
  background: rgb(61 71 83 / 75%);
  backdrop-filter: blur(15px);
  border: 1px solid rgb(151 151 151 / 30%);
  border-radius: 10px;
}

.avatar {
  background-color: rgb(var(--v-theme-cockpit-highlight));
  border-radius: 15px;
}
</style>
