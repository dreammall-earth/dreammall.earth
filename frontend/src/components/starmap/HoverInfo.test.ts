import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach } from 'vitest'

import HoverInfo from './HoverInfo.vue'

import type { UserWithProfile } from '#stores/userStore'

describe('HoverInfo', () => {
  const Wrapper = (
    props = {
      data: {
        id: 1,
        name: 'Marianne Müller',
        username: 'username',
        availability: 'available' as const,
        details: [],
        social: [],
      } as UserWithProfile | null,
      x: 100,
      y: 200,
      showMoreButton: true,
    },
  ) => {
    return mount(HoverInfo, {
      props,
    })
  }
  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders without Show More button', () => {
    wrapper = Wrapper({
      data: {
        id: 1,
        name: 'Marianne Müller',
        username: 'username',
        availability: 'available' as const,
        details: [],
        social: [],
      },
      x: 100,
      y: 200,
      showMoreButton: false,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders without star data', () => {
    wrapper = Wrapper({
      data: null,
      x: 100,
      y: 200,
      showMoreButton: false,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('show more button', () => {
    it('emits show-more event', async () => {
      await wrapper.find('button').trigger('click')
      expect(wrapper.emitted('show-more')).toBeTruthy()
    })
  })
})
