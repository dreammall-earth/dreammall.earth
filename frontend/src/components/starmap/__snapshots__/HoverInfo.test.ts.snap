// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HoverInfo > renders 1`] = `
<div
  class="hover-info"
  data-v-899f618d=""
  style="left: 50px; top: 150px;"
>
  <div
    class="info-box pa-2"
    data-v-899f618d=""
  >
    <div
      class="v-avatar v-theme--light v-avatar--density-default v-avatar--variant-flat avatar mb-2 text-font"
      data-v-899f618d=""
      style="width: 25px; height: 25px;"
    >
      
      
      <span
        data-v-899f618d=""
      >
        MM
      </span>
      
      
      
      <!---->
      <span
        class="v-avatar__underlay"
      />
      
    </div>
    <h3
      data-v-899f618d=""
    >
      Marianne Müller
    </h3>
  </div>
  <div
    class="mt-2 d-flex align-center justify-center"
    data-v-899f618d=""
  >
    <button
      data-v-899f618d=""
    >
      <svg
        data-v-899f618d=""
        fill="none"
        height="28"
        viewBox="0 0 28 28"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-v-899f618d=""
          opacity="0.75"
        >
          <circle
            cx="14"
            cy="14"
            data-v-899f618d=""
            fill="#3D4753"
            r="14"
          />
          <circle
            cx="14"
            cy="14"
            data-v-899f618d=""
            r="13.5"
            stroke="#979797"
            stroke-opacity="0.3"
          />
        </g>
        <circle
          cx="7"
          cy="14"
          data-v-899f618d=""
          fill="#CCCCCC"
          r="2"
          transform="rotate(-90 7 14)"
        />
        <circle
          cx="14"
          cy="14"
          data-v-899f618d=""
          fill="#CCCCCC"
          r="2"
          transform="rotate(-90 14 14)"
        />
        <circle
          cx="21"
          cy="14"
          data-v-899f618d=""
          fill="#CCCCCC"
          r="2"
          transform="rotate(-90 21 14)"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`HoverInfo > renders without Show More button 1`] = `
<div
  class="hover-info"
  data-v-899f618d=""
  style="left: 50px; top: 150px;"
>
  <div
    class="info-box pa-2"
    data-v-899f618d=""
  >
    <div
      class="v-avatar v-theme--light v-avatar--density-default v-avatar--variant-flat avatar mb-2 text-font"
      data-v-899f618d=""
      style="width: 25px; height: 25px;"
    >
      
      
      <span
        data-v-899f618d=""
      >
        MM
      </span>
      
      
      
      <!---->
      <span
        class="v-avatar__underlay"
      />
      
    </div>
    <h3
      data-v-899f618d=""
    >
      Marianne Müller
    </h3>
  </div>
  <!--v-if-->
</div>
`;

exports[`HoverInfo > renders without star data 1`] = `
<div
  class="hover-info"
  data-v-899f618d=""
  style="left: 50px; top: 150px;"
>
  <div
    class="info-box pa-2"
    data-v-899f618d=""
  >
    <div
      class="v-avatar v-theme--light v-avatar--density-default v-avatar--variant-flat avatar mb-2 text-font"
      data-v-899f618d=""
      style="width: 25px; height: 25px;"
    >
      
      
      <span
        data-v-899f618d=""
      />
      
      
      
      <!---->
      <span
        class="v-avatar__underlay"
      />
      
    </div>
    <h3
      data-v-899f618d=""
    />
  </div>
  <!--v-if-->
</div>
`;
